### PR TITLE
refactor: interested-party ni integration

### PR DESCRIPTION
**Pull request by JamesP-KC**
## Describe your changes

ASB-1468

Refactoring work ahead of implementing new functionality in ASB-1468.

Overview of changes:
- The `PUT /interested-party/{id}/comments` and `POST /interested-party/{token}` endpoints have been removed. 
- These were used to support a previous design where a user could start registering, save their progress and come back later ("draft")
- Our current design does not make use of the `/{token}` endpoint at all so can be safely removed without being a breaking change.
- Removing the `PUT /comments` endpoint is a breaking change. The current implementation on web invokes the POST to create an InterestedParty entity, and then immediately invokes the `PUT /comments` endpoint to update the created InterestedParty with a comment. We can streamline this process by merging those two API calls into one: a POST with the `comment` included. The webapp code has been updated to do this.
- Amend the `POST /interested-party` endpoint's response to return JSON (`{ "referenceId": "1234" }`) rather than a string `"1234"`.

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [x] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

**Review by Yakitrak — APPROVED** [review]
Approved - don't know much about of this part of the codebase, but everything looks good to me based on what your PR description. A small comment on unused vars.

File: packages/applications-service-api/__tests__/unit/controllers/interested-party.test.js [hunk 1]
```diff
@@ -58,6 +56,7 @@ const ipDataOwnBehalf = {
 	web_ref: 1
 };
 
+// eslint-disable-next-line no-unused-vars
```
[the remark this answers is not in the document]
Comment (JamesP-KC) [reply packages/applications-service-api/__tests__/unit/controllers/interested-party.test.js:59]: These vars are unused but I was planning to use them in my next commit for the functionality, so just disabled the lint check. As it happens, I've relocated these vars to another file so for this PR I will just remove the lint bypass

**Review by Philip-Philippou1 — APPROVED** [review]
This is so much cleaner.  Would be worth running the e2e tests on this as they are pretty detailed.

**Pull request merged**